### DISCUSSION TITLE
Output created but NO preview?

### DISCUSSION THREAD

**Discussion by DarvidTorres** [quarto-dev/quarto-cli#3749] [Q&A]
I'm running the Hello, Quarto tutorial for VSCode on Windows 10. It seems that everything's working, except that I see no preview. I checked the Quarto extension setting for "preview" to "internal" and checked the "reveal the preview panel..." as default. But I see no preview whatsoever. Here's my terminal:

PS C:\Programming\R\R Programming A-Z> quarto preview "c:/Programming/R/R Programming A-Z/hello.qmd" --to html --no-browser --no-watch-inputs

Starting python3 kernel...Done

Executing 'hello.ipynb'
  Cell 1/1...Done

pandoc 
  to: html
  output-file: hello.html     
  standalone: true
  section-divs: true
  html-math-method: mathjax   
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Quarto Basics
  jupyter: python3

Output created: hello.html

Watching files for changes

**Comment by cscheid** [comment]
You're using `--no-browser`, which instructs quarto not to open a web browser. So it doesn't. (I don't understand why you're not seeing the prompt which tells you the web server URL, though.) Try running without that option.

**Reply by DarvidTorres** [reply]
Thanks for answering! The `--no-browser` flag is the default option when rendering (and I don't know how to change it). Though there's no preview after runing it again without the option. There's no preview either rendering to PDF. Maybe it's a VSCode setting that I'm missing? I really want to preview my work within VSCode but don't know how

![image](https://user-images.githubusercontent.com/20685823/209562080-35019c1d-5b6a-49c5-96b6-8bf45340f712.png)

![image](https://user-images.githubusercontent.com/20685823/209562214-80092921-e07a-43a3-b050-04ff4085f15d.png)

**Reply by cscheid** [reply]
I don't understand what's going on; on OS X, I can't reproduce your issue. @cderv, do you have any insights here?

**Reply by milanmlft** [reply]
Same thing happens for me on macOS, when I run Quarto preview through one of the VSCode shortcuts, it runs with the `--no-browser` and `--no-watch-inputs` flags.

**Comment by DarvidTorres** [comment]
The issue is that there's no preview panel in VSCode when I render the file. 

This is what the tutorial page looks like:

![image](https://user-images.githubusercontent.com/20685823/209586880-0df586d9-f9b7-4450-9169-8f954651327b.png)

This is mine:

![image](https://user-images.githubusercontent.com/20685823/209586963-3f31908d-cb94-41fd-9663-90d813ae68d1.png)

The difference is that there's no preview panel in my vscode after rendering, even without the `--no-browser` flag and I don't know how to open that preview panel.

**Reply by cscheid** [reply]
It's definitely weird that you're seeing this. Any of these work for me:

- `Shift-Alt-K` shortcut
- `Shift-Alt-P` and then find "Quarto: Render" on the command bar
- Click on the `render` button on the .qmd tab:
   <img width="610" alt="image" src="https://user-images.githubusercontent.com/285675/209680790-2fd76c77-89f0-426a-9f70-135f8e750af2.png">

Is it possible that some other VS code extension installed that is interfering?

**Comment by kvten** [comment]
My output is different from what you have. (see screenshot below) There's a local host link I can open in browser in addition to preview panel.

![Screenshot](https://user-images.githubusercontent.com/103538771/209726882-869e1715-5956-4a0b-9000-62f214438b6f.png)

- VS Code v1.74.2
- Quarto v1.59.0

**Reply by DarvidTorres** [reply]
Hi kenf1, can you tell me what's the command you're running? I'm using `quarto preview "c:/folder/file.qmd" --to html` as shown in the screenshots above, maybe you're employing a different flag? Thanks

**Reply by kvten** [reply]
I have a keyboard shortcut in VS Code set to `quarto.render`. My quarto settings (in VS Code):

![Screenshot](https://user-images.githubusercontent.com/103538771/209735335-ae615474-2137-47ef-9488-58168b48617e.png)

**Reply by DarvidTorres** [reply]
I have the same settings as you do and still can see the "Browser at ... " leyend after "watching files for changes" output.

**Reply by half-normal** [reply]
I was having this same problem after updating Quarto. A pre-existing quarto website would not render properly from the vs-code ctrl+shft+p commands nor the 'render' button. It would render correctly from the terminal, but preview would be sent to the browser. Meanwhile a new quarto website would render and preview correctly.

If this is the case for you, try deleting the _extensions, .quarto and _site/docs (i.e. the rendered website) folders and then reinstall the extensions. Now try rendering. This fixed the problem for me, but I'm not sure why.

EDIT: This happened again, deleting and reinstalling the extensions fixed it again. I wonder if this an issue with the fontawesome extension as this is the only one I am currently using.

**Comment by arthurgailes** [comment]
Also having this problem; it works fine in RStudio, and the localhost output renders in a browser, but the render on save function hangs as well. 

Notes:

- if I force quit (ctrl+c in the terminal), then the internal viewer pops up, empty, and I get a successfully rendered message.
- it works fine with the external browser, including the render-on-save refresh

**Comment by allefeld** [comment]
This seems to be happening again. No problem until yesterday; I installed the newest Quarto version, but after that everything continued to work.

Today, opening the same folder & documents, and rendering via Ctrl-Shift-K, the internal preview does not open. In the Quarto terminal it says that the output has been created and I can browse it at a localhost address. But the "Rendering" message does not stop, and saving the document does not trigger a re-render. I can interrupt the rendering process via Ctrl-C in the Quarto terminal, but nothing happens but returning to the shell prompt.

Command created by the Quarto extension:
```
quarto preview <path>/test-document.qmd --no-browser --no-watch-inputs
```

![image](https://user-images.githubusercontent.com/1738353/230740214-58db95ac-3d6c-4908-a2b9-a1be38f358fc.png)

VS Code 1.77.1, Quarto Extension 1.79.0, Quarto 1.3.313
Debian 11.6, kernel 5.10.0-21-amd64

**Edit:** ~~This seems only to happen when a folder is open in which a Quarto extension is edited.~~ Weirdly, editing & saving the `_extension.yml` file does lead to a re-render (but still no internal preview).

**Edit 2:** Ok, I think I figured it out. It happened for me when I was editing a Quarto format extension, but it's not really caused by that. I had created a document showcasing Pandoc & Quarto features, to test my Quarto format extension. Part of that was a Python code block which queried Quarto's built-in Pandoc for a list of the languages for which syntax highlighting is supported. Here is a minimal version:

``````
```{python}
import subprocess

subprocess.run(['quarto', 'pandoc', '--list-highlight-languages'],  stderr=subprocess.PIPE)
```
``````

Create a `qmd` file with this code block, and it should trigger the bug.

It  seems that calling `quarto` in a subprocess, while it is already running from the VS Code Quarto extension, upsets things. If I directly call `/opt/quarto/bin/tools/pandoc`, it doesn't happen. If I call independently installed Pandoc, it doesn't happen either.

This is a rather exotic bug which will hardly affect anyone. Let me know if you'd like me to open a separate issue.

**Comment by stewartli** [comment]
inside of my vscode:
1. click render button -> no preview -> I guess no browser is the default.
2. key in terminal -> preview shows up.

**Comment by Kkhurana007** [comment]
I'm having the same issue.
Win 10 professional X64
IDE- Visual studio code
quarto version- 1.3.340

first, i rendered. The pdf gets generated, but does not open-
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/bf3e5b09-1cf4-4a73-b997-9bf1479f6cc1)

then, I previewed-
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/e1c3e6e1-cfb2-49b9-aa55-e6a9e7f2e9c8)
and got the following error-
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/51e8dbca-0e2e-445b-a44e-3732745a6405)

Anybody, please help?

**Reply by mcanouil** [reply]
Could you update to at least use the latest stable version?
Could you try a very simple document? Try with html then PDF.

````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

![A placeholder image](https://placehold.co/600x400.png)

The end.
````

**Reply by Kkhurana007** [reply]
The .html file works fine- https://ilovesoils.com/posts/pythonic_thinking/Pyhton%20Coding%20Rules.html 

I just wanted to have a pdf version of it as well.

Thanks for trying to help me with this!

**Reply by cscheid** [reply]
@Kkhurana007 Can you provide a single .qmd file that causes the problem for you?

**Reply by Kkhurana007** [reply]
Unfortunately, I can't upload the .ipynb file, but here is the link to that file https://ulavaldti-my.sharepoint.com/:u:/g/personal/kukhu1_ulaval_ca/EctJmLhWa2tBja8ly-PqGDUB4oDT2x6KeLyzttZ79bf-dw?e=9ZLhls  

Many thanks @cscheid

**Reply by cscheid** [reply]
I get a `403: forbidden` error message when trying to access that.

**Reply by Kkhurana007** [reply]
yes, it's .ipynb file. Can be opened with IDE like jupyter, visual studio code, etc.

**Reply by cscheid** [reply]
I cannot _download_ the file.

**Reply by Kkhurana007** [reply]
sorry about that. please try now https://ulavaldti-my.sharepoint.com/:u:/g/personal/kukhu1_ulaval_ca/EctJmLhWa2tBja8ly-PqGDUB4oDT2x6KeLyzttZ79bf-dw?e=Lco3Oe

**Reply by cscheid** [reply]
It's the same error. Please confirm that you yourself can download it from an incognito window before trying to give me another link again.

**Reply by Kkhurana007** [reply]
kindly try with this one - https://github.com/Kkhurana007/fr_blog_mine/blob/main/posts/pythonic_thinking/Coding_Rules.ipynb

**Reply by cscheid** [reply]
It really is something with your pdflatex setup - `quarto render Coding_Rules.ipynb --to pdf` produces this for me:

<img width="755" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/2d980d0d-59d3-4c48-a1cf-a5633ae8b8a3">

**Reply by Kkhurana007** [reply]
I succeeded finally. I can see it in IDE, but the pdf doesn't open in Adobe or Microsoft's built-in pdf reader. 
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/549ea925-88bc-4c7c-9add-c2abea413e23)
Any insights?

**Reply by cderv** [reply]
> but the pdf doesn't open in Adobe or Microsoft's built-in pdf reader.

You mean automatically when doing a preview ? Or you can't open it when you double click on the file ?

**Comment by Kkhurana007** [comment]
when I double-click on the file. please see attached!
[Coding_Rules.pdf](https://github.com/quarto-dev/quarto-cli/files/13635018/Coding_Rules.pdf)

**Reply by cderv** [reply]
When I try to render the ipynb shared there: https://github.com/quarto-dev/quarto-cli/discussions/3749#discussioncomment-7801965

it works ok for and produce a PDF I can open outside of the IDE. 

I don't know how yours got corrupted. You should check your LaTeX distribution that produce the PDF. Be sure it is up to date;

**Reply by Kkhurana007** [reply]
I think it has something to do with .yml settings. The LaTex files are up to date as the pdf documents work fine in other projects.
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/68da2916-23ba-440b-a4d3-e9822188fc87)
Here is the link- https://github.com/Kkhurana007/fr_blog_mine/blob/main/_quarto.yml

**Reply by cderv** [reply]
Also using this YAML for pdf format, I can render the document ok. 

I also cloned you all repo and ran 
````
quarto render .\posts\pythonic_thinking\Coding_Rules.ipynb
````

which successfuly produced a HTML file and a PDF file that I can open.

**Reply by Kkhurana007** [reply]
Perfect. I'll figure it out then. Many thanks for your time!
